include/seqan3/search/configuration/all.hpp

```diff
@@ -163,4 +163,20 @@
  * The seqan3::search_cfg::parallel configuration element can be combined with any other search configuration.
  *
  * \include test/snippet/search/configuration_parallel.cpp
+ *
+ * ### User callback
+ *
+ * In the default case, a call to seqan3::search returns a lazy range over the results of the search. This lazy range
+ * has the advantage that the results are always in a deterministic order even if the search is executed in parallel.
+ * Sometimes, however, it might be challenging to provide a user defined callback.
```

Comment:
```suggestion
 * Sometimes, however, it might be preferred to provide a user defined callback.
```
? Why should it be challenging

Reply:
Ask @eseiler 😏  I had demanding.

Reply:
demanding in this context means `difficult`; but obviously you meant preferred, I just didn't catch your meaning :)

Reply:
Ok, what I actually meant was desirable

Reply:
:smile: desirable it is